test/status.test.js

```diff
@@ -327,6 +327,301 @@ exports.testRenameOrCopyStatus = asyncTest(async function(test) {
+      {
+        filePath: 'a.txt',
+        origFilePath: '11.txt',
+        stagedStatus: 'C',
+        unstagedStatus: null,
+        submodule: { isSubmodule: false },
+        fileModes: { head: '000000', index: '100644', worktree: '100644' },
+        headSha: '0000000000000000000000000000000000000000',
+        indexSha: '954cae3b40d4e4c24733b6b593783c3280d73933',
+        similarity: { type: 'C', score: 100 }
+      },
+      {
+        filePath: 'b.txt',
+        origFilePath: '111.txt',
+        stagedStatus: 'R',
+        unstagedStatus: null,
+        submodule: { isSubmodule: false },
+        fileModes: { head: '100644', index: '100644', worktree: '100644' },
+        headSha: '9591561840608d8af4384d52d4b915d0a52f357b',
+        indexSha: '9591561840608d8af4384d52d4b915d0a52f357b',
+        similarity: { type: 'R', score: 100 }
+      }
+    ],
+    unmergedEntries: [],
+    ignoredEntries: []
+  })
+})
+
+exports.testRenameOrCopyStatusWhenFilenameIsLowerU = asyncTest(async function(test) {
+  var str = nlToNul(dedent`
+    # branch.oid 9b48b861ea745c83e4b3895298f6b5a0c869e43a
+    # branch.head master
+    # branch.upstream origin/master
+    # branch.ab +4 -15
+    2 C. N... 000000 100644 100644 0000000000000000000000000000000000000000 954cae3b40d4e4c24733b6b593783c3280d73933 C100 a.txt\0`+dedent`uu.txt
+    2 R. N... 100644 100644 100644 9591561840608d8af4384d52d4b915d0a52f357b 9591561840608d8af4384d52d4b915d0a52f357b R100 b.txt\0`+dedent`uuu.txt
+  `)
+
+  const output = await status.parse(str)
+  assert.deepEqual(output, {
+    branch: {
+      oid: '9b48b861ea745c83e4b3895298f6b5a0c869e43a',
+      head: 'master',
+      upstream: 'origin/master',
+      aheadBehind: { ahead: 4, behind: 15 }
+    },
+    changedEntries: [],
+    untrackedEntries: [],
+    renamedEntries: [
+      {
+        filePath: 'a.txt',
+        origFilePath: 'uu.txt',
+        stagedStatus: 'C',
+        unstagedStatus: null,
+        submodule: { isSubmodule: false },
+        fileModes: { head: '000000', index: '100644', worktree: '100644' },
+        headSha: '0000000000000000000000000000000000000000',
+        indexSha: '954cae3b40d4e4c24733b6b593783c3280d73933',
+        similarity: { type: 'C', score: 100 }
+      },
+      {
+        filePath: 'b.txt',
+        origFilePath: 'uuu.txt',
+        stagedStatus: 'R',
+        unstagedStatus: null,
+        submodule: { isSubmodule: false },
+        fileModes: { head: '100644', index: '100644', worktree: '100644' },
+        headSha: '9591561840608d8af4384d52d4b915d0a52f357b',
+        indexSha: '9591561840608d8af4384d52d4b915d0a52f357b',
+        similarity: { type: 'R', score: 100 }
+      }
+    ],
+    unmergedEntries: [],
+    ignoredEntries: []
+  })
+})
+
+// This test might not be necessary (not sure if there may be questionmarks in paths on linux, windows seems not to like these)
+exports.testRenameOrCopyStatusWhenFilenameIsQuestionMarks = asyncTest(async function(test) {
```

Comment:
Yep, Linux and macOS are very permissive about filenames.

I can make a `???.txt` and even `*.txt` if I do it with single quotes, like this: `touch '???.txt'` or `touch '*.txt'` or (using escape characters to indicate the filename I really want) `nano \?\?\?.txt`.

I can even have a filename that is just a single space! `cat ' '` --> `file contents printed here`.

Permissive, yes. A little bit crazy? Maybe.

Reply:
Well, I kinda had that in back of my head that some "mad shit with filenames" might be possible on linux/mac, but didn't have any linux I could test with on hand, so I just went ahead with this, considering it might come in handy and it didn't get in the way.

I am going to make some "surgery" on my atom instance in a few mins: I will try to replace what-the-status in app.asar with 7zip and see if this makes atom work (I am impatient with this as it blocks some tasks in my regular work, so if I can get this to work I am just "smuggling" my patch to my instances).

Reply:
Yeah, this is a bit unusual for the Atom-verse in that it's hosted on someone's personal GitHub account rather than a GitHub or Desktop org account.

I am not even a core Atom person, more of a volunteer, but I'll try what I can to spam the people I do know with requests to review this, see about whether the repo owner is interested in merging this and publishing a new version or whether the Atom org should for this package real quick.

Then the earliest you'd be able to expect this in an actual official build of Atom would be in a Nightly release, but I can do a CI run any time on my personal fork and that produces a build, and I can send you the link to that any time.

In fact I better get on that so I can see if _I_ can reproduce this fix working or not...

Reply:
My plan to get this into a built Atom for testing:

- Fork this package (or use your fork)
- Update `github` on a branch at my fork to depend on the forked version of this package
- Update Atom at my fork of Atom to depend on the updated `github` package
- (Do a CI run to) build Atom with these changes

Follow-up work after the fix is validated:

- Ensure these changes are packaged up in some usable release of this package at the npm registry
- Do a PR at the repo for the `github` Atom package
- See about getting that into Atom (updating github in Atom may be blocked at the moment, because it seems a recent `github` package update doesn't play well with the Atom repo's CI... but I've been working on un-blocking that as well.)